system/lib/wasmfs/file.h

```diff
@@ -255,6 +257,19 @@ class Directory : public File {
     }
   }
 };
+
+struct ParsedPath {
+  std::optional<Directory::Handle> parent;
+  std::shared_ptr<File> child;
+};
+
+// TODO: Should this return a locked parent handle or a pointer to the parent?
+// Given a pathname, this function will return a locked parent directory and a
+// pointer to the specified file.
```

Comment:
After we redo the locking on the directory structure, these will definitely both be normal shared pointers, but until then having the parent locked makes sense.

Reply:
The reason for this comment was that currently `rename` should not pre-emptively lock the parent, since we want to trylock on it.